packages/form-upload/Upload.js

```diff
@@ -37,12 +54,74 @@ const Upload = ({
 
   const fieldValue = Array.isArray(field.value) ? field.value : [];
 
+  const callFileDelivery = useCallback(
+    async upload => {
+      if (!Array.isArray(upload)) upload = [upload];
+      const uploadResults = [];
+      try {
+        for (const u of upload) {
+          const data = {
+            deliveries: [
+              {
+                deliveryChannel,
+                fileURI: u.references[0],
+                metadata:
+                  typeof fileDeliveryMetadata === 'function'
+                    ? fileDeliveryMetadata(u)
+                    : fileDeliveryMetadata,
+              },
+            ],
+          };
+
+          uploadResults.push(
+            avFilesDeliveryApi.uploadFilesDelivery(data, {
+              clientId,
+              customerId,
+            })
+          );
+        }
+
+        await Promise.all(uploadResults);
+      } catch (error) {
+        setFieldError(name, 'An error occurred while uploading files.');
+      }
+    },
+    [
+      clientId,
+      customerId,
+      deliveryChannel,
+      fileDeliveryMetadata,
+      name,
+      setFieldError,
+    ]
+  );
+
+  async function checkValidFormAndCallFileDelivery() {
+    if (Object.keys(errors).length === 0) {
+      await callFileDelivery(fieldValue);
+    }
+  }
+
+  useEffect(() => {
+    if (
+      !onFileUpload &&
+      isSubmitting === true &&
+      isValidating === false &&
+      deliverFileOnSubmit &&
+      deliveryChannel &&
+      fileDeliveryMetadata
+    ) {
+      checkValidFormAndCallFileDelivery();
+    }
+    // FIXME: add all dependencies to the array and add logic to only submit upload once
```

Comment:
is this plan for this to be fixed in this or a follow up PR? what's the complication?

Reply:
I think when `fileDeliveryMetadata` is included as a dependency that its reference continually changes and causes the effect hook to rerun. The issue was we were seeing multiple API calls when including all the suggested dependencies, I can address this before merging, just wanted to make sure we weren't missing edge cases by excluding values from the dependency array